### PR TITLE
Fix private mode browsing by falling back to browser cookie.

### LINKED ISSUE #225: QuotaExceededError on iOS 7 or Safari Private Mode.
cvz1adawaaaccvd

### PR DESCRIPTION
**Pull request by harshavardhana**
Fixes #225

### CODE REVIEW

**Comment by harshavardhana** [comment]
ping @rushenn @krishnasrinivas - can you guys help me validate this patch? iOS Safari private browsing mode - Thanks.

**Comment by krishnasrinivas** [comment]
@harshavardhana got this error on the console:
```
store.js:81 Uncaught ReferenceError: data is not defined
```

**Comment by harshavardhana** [comment]
> store.js:81 Uncaught ReferenceError: data is not defined

Fixed @krishnasrinivas @rushenn

**Comment by krishnasrinivas** [comment]
@harshavardhana the code flow in store.js was hard to follow, mainly because the store() function was doing function overloading. Instead checkout https://github.com/ripeworks/local-storage-fallback, the API is nice and intuitive, code is also simpler.

**Comment by harshavardhana** [comment]
> @harshavardhana the code flow in store.js was hard to follow, mainly because the store() function was doing function overloading. Instead checkout https://github.com/ripeworks/local-storage-fallback, the API is nice and intuitive, code is also simpler.

So does this approach works is what i wanted to know if yes then we can use any library which does a better job. @krishnasrinivas @rushenn ?

**Comment by krishnasrinivas** [comment]
> So does this approach works is what i wanted to know if yes then we can use any library which does a better job. @krishnasrinivas @rushenn ?

should work, rushenn had trouble viewing his laptop IP from his iphone, I will upload a minio binary with this change to m1.minio.io and check if rushenn can access it fine on his iphone.

**Comment by harshavardhana** [comment]
> should work, rushenn had trouble viewing his laptop IP from his iphone, I will upload a minio binary with this change to m1.minio.io and check if rushenn can access it fine on his iphone.

@rushenn were you able to validate on your iphone?

**Comment by rushenn** [comment]
@harshavardhana no, I'm having trouble viewing on mobile. Discussed with Krishna, I will validate once m1.minio.io is live

**Comment by harshavardhana** [comment]
> @harshavardhana no, I'm having trouble viewing on mobile. Discussed with Krishna, I will validate once m1.minio.io is live

Updated this patch to use the package @krishnasrinivas mentioned.

**Comment by krishnasrinivas** [comment]
let me compile and upload the binary to m1

**Comment by krishnasrinivas** [comment]
@harshavardhana getting this in console:

```
index_bundle-2016-11-06T19-44-26Z.js:29 Uncaught TypeError: Cannot assign to read only property 'localStorage' of object '#<Window>'
```

also we'll have to do storage.getItem() setItem() and removeItem() to work seamlessly https://github.com/ripeworks/local-storage-fallback/blob/master/src/CookieStorage.js

**Comment by harshavardhana** [comment]
> also we'll have to do storage.getItem() setItem() and removeItem() to work seamlessly https://github.com/ripeworks/local-storage-fallback/blob/master/src/CookieStorage.js

Ah the shim doesn't work. Let me use the getItem()

**Comment by krishnasrinivas** [comment]
@rushenn i have messaged you the access details to verify from safari brivate browsing. you can confirm if it works here.

**Comment by krishnasrinivas** [comment]
@rushenn confirmed : `It's working I was able to login and create bucket without any issues in private mode`
